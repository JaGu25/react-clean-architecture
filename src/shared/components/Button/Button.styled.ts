import styled from 'styled-components'
import { colors, fontSizes, spacing } from '../../utils/styles.utils'

interface ButtonWrapperProps {
    color: string,
    color_variant: string
}

const ButtonWrapper = styled.button<ButtonWrapperProps>`
    padding: ${spacing.Spacing1} ${spacing.Spacing2};
    background-color: ${({ color }) => `${color}`};
    font-size: ${fontSizes.sm};
    color: ${colors.white};
    border-radius: ${spacing.Spacing1};
    cursor: pointer;
    transition-duration: 0.2s;

    &:hover {
        background-color: ${({ color_variant }) => `${color_variant}`};
    }

    &:active {
        transform: translateY(2px);
    }
`

const ButtonStyled = {
    ButtonWrapper
}

export default ButtonStyled
