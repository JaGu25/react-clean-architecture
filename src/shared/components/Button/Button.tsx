import React from 'react'
import ButtonStyled from './Button.styled'
import { colors } from '../../utils/styles.utils'

interface IProps {
    text: string,
    actionType?: 'button' | "reset" | "submit"
    type?: 'primary' | 'secondary'
    onClick: () => void
}

const Button: React.FC<IProps> = ({
    text,
    actionType = "button",
    type = "primary",
    onClick
}) => {
    return (
        <ButtonStyled.ButtonWrapper
            type={actionType}
            color={colors[type]}
            color_variant={colors[`${type}_variant`]}
            onClick={onClick}
        >
            {text}
        </ButtonStyled.ButtonWrapper>
    )
}

export default Button
