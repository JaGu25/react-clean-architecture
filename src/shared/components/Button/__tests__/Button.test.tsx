import React from "react"
import { fireEvent, render, screen } from "@testing-library/react"
import Button from "../Button"

const click = jest.fn()

describe('Button Component', () => {
    it('should render without crash', () => {
        const { container } = render(
            <Button
                text="click me"
                onClick={click}
            />
        )
        expect(container).toMatchSnapshot()
    })

    it('should click button', async () => {
        render(
            <Button
                text="click me"
                onClick={click}
            />
        )

        const button = await screen.findByText('click me')
        fireEvent.click(button)
        expect(click).toBeCalled()
    })
})