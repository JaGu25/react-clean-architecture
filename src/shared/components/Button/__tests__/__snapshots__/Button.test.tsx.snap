// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render without crash 1`] = `
<div>
  <button
    class="sc-bczRLJ fFZzDd"
    color="#6200EE"
    type="button"
  >
    click me
  </button>
</div>
`;
